docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,322 @@
+# Downward APIs for resource limits and requests
+
+## Background
+Currently downward APIs (via environment variables and volume plugin) only support exposing a Pod's name, namespace, annotations, labels and its IP. This document explains the need and design to extend them to expose resources (e.g. cpu, memory, storage) limits and requests.
+
+## Motivation
+By exposing information about computing resource requests & limits into pods, image/application authors can be aware of the computing environment in which the pods run, and could modify container images accordingly.
+
+## Design
+This is mostly driven by the discussion in https://github.com/kubernetes/kubernetes/issues/9473.
+
+### Downward APIs via environment variables and volume plugin
```

Comment:
This is just a top level section for all the discussion below. I can remove if it does not make sense.